### PR TITLE
Fixed issue with barrier fetch operation with batch consume

### PR DESCRIPTION
**Pull request by pranavrth**

### CODE REVIEW

File: tests/0134-purge_same_partition_messages_after_seek_in_same_batch_consume.c [hunk 1]
```diff
@@ -0,0 +1,281 @@
+/*
```
Comment (edenhill) [inline tests/0134-purge_same_partition_messages_after_seek_in_same_batch_consume.c:1]: Following the repo style, this filename is way too long and is more of a test description (which goes in the test itself).
Suggest renaming to 0136-toppar_barrier.c.
Use number 0136:  134 and 135 are taken in other PRs.

File: src/rdkafka_queue.c [hunk 2]
```diff
@@ -548,8 +549,8 @@ static size_t rd_kafka_purge_outdated_messages(int32_t version,
         for (i = 0; i < cnt; i++) {
                 rd_kafka_op_t *rko;
                 rko = rkmessages[i]->_private;
-                if (rd_kafka_op_version_outdated(rko, version)) {
-                        /* This also destroys the corresponding rkmessage. */
+                if (rko->rko_rktp == rktp &&
+                    rd_kafka_op_version_outdated(rko, version)) {
```
Comment (edenhill) [inline src/rdkafka_queue.c:553]: Looking at op_version_outdated(), I think we might want to call it version set to 0 and let it just fall down to its 3rd if-statement:
https://github.com/edenhill/librdkafka/blob/master/src/rdkafka_partition.h#L806

File: tests/test.c [hunk 3]
```diff
@@ -480,6 +481,7 @@ struct test tests[] = {
     _TEST(0131_connect_timeout, TEST_F_LOCAL),
     _TEST(0132_strategy_ordering, 0, TEST_BRKVER(2, 4, 0, 0)),
     _TEST(0133_ssl_keys, TEST_F_LOCAL),
+    _TEST(0134_purge_same_partition_messages_after_seek_in_same_batch_consume, 0, TEST_BRKVER(2, 4, 0, 0)),
```
Comment (edenhill) [inline tests/test.c:484]: Why does it require broker version >= 2.4?

**Comment by pranavrth** [comment]
Fix as part of https://github.com/confluentinc/librdkafka/pull/4117

**Pull request closed without merging**